superset-frontend/src/dashboard/components/RefreshIntervalModal.tsx

```diff
@@ -57,8 +43,31 @@ const RefreshWarningContainer = styled.div`
   margin-top: ${({ theme }) => theme.gridUnit * 6}px;
 `;
 
-class RefreshIntervalModal extends React.PureComponent {
-  constructor(props) {
+type RefreshIntervalModalProps = {
+  triggerNode: JSX.Element;
+  refreshFrequency: number;
+  onChange: (refreshLimit: number, editMode: boolean) => void;
+  editMode: boolean;
+  refreshLimit?: number;
+  refreshWarning?: string;
```

Comment:
same comments on this not needing to be undefined (and since refreshWarning can be null, this should be `refreshWarning: string | null`